[APM] Cannot click through to details of an error on the waterfall page if the error is missing `transaction.id`

cauemarcondes opened the pull request:
closes #66363

<img width="599" alt="Screenshot 2020-05-13 at 14 13 48" src="https://user-images.githubusercontent.com/55978943/81810893-0c8ab680-9524-11ea-89f3-679893df82e3.png">

cauemarcondes committed:
removing transaction from link when it isnt defined

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

sqren commented on x-pack/plugins/apm/public/components/shared/charts/Timeline/Marker/ErrorMarker.test.tsx:
```diff
@@ -0,0 +1,77 @@
+      serviceColor: '#fff'
+    } as ErrorMark;
+    const component = shallow(<ErrorMarker mark={mark} />);
+    expect(component).toMatchSnapshot();
```
Instead of a snapshot can you get the link `kuery` from the link, like:

```
expect(decodeURIComponent(kuery)).toBe('trace.id : "123" and transaction.id : "456"')
```

cauemarcondes committed:
refacroting test

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/)
* Commit: 9268f3d436cb8e0835e9a1bf882f20d37c3e2617

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/","number":"47609","commit":"9268f3d436cb8e0835e9a1bf882f20d37c3e2617"}]}
PIPELINE-->

cauemarcondes committed:
refacroting test

cauemarcondes commented:
@sqren can you check it again please? I've changed the way I expecting the URL, didn't like to previous. And I also removed the `encodeURIComponent` from `ErrorMarker` because it's already being encoded on `APMLink`.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47663/)
* Commit: ba5c464a438637aec66159031c4a4e76e45467e6

### History
* :green_heart: [Build #47609](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/) succeeded 9268f3d436cb8e0835e9a1bf882f20d37c3e2617

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/","number":"47609","commit":"9268f3d436cb8e0835e9a1bf882f20d37c3e2617"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47663/","number":"47663","commit":"ba5c464a438637aec66159031c4a4e76e45467e6"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47693/)
* Commit: 55e63ce90801f082e5c459daedff2473c9fab8ea

### History
* :green_heart: [Build #47663](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47663/) succeeded ba5c464a438637aec66159031c4a4e76e45467e6
* :green_heart: [Build #47609](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/) succeeded 9268f3d436cb8e0835e9a1bf882f20d37c3e2617

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47609/","number":"47609","commit":"9268f3d436cb8e0835e9a1bf882f20d37c3e2617"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47663/","number":"47663","commit":"ba5c464a438637aec66159031c4a4e76e45467e6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47693/","number":"47693","commit":"55e63ce90801f082e5c459daedff2473c9fab8ea"}]}
PIPELINE-->
